Sources/System/Internals/Syscalls.swift

```diff
@@ -118,20 +118,90 @@ internal func system_dup2(_ fd: Int32, _ fd2: Int32) -> Int32 {
   return dup2(fd, fd2)
 }
 
-#if !os(Windows)
 internal func system_pipe(_ fds: UnsafeMutablePointer<Int32>) -> CInt {
 #if ENABLE_MOCKING
   if mockingEnabled { return _mock(fds) }
 #endif
   return pipe(fds)
 }
-#endif
 
-#if !os(Windows)
 internal func system_ftruncate(_ fd: Int32, _ length: off_t) -> Int32 {
 #if ENABLE_MOCKING
   if mockingEnabled { return _mock(fd, length) }
 #endif
   return ftruncate(fd, length)
 }
+
+internal func system_mkdir(
+    _ path: UnsafePointer<CInterop.PlatformChar>,
+    _ mode: CInterop.Mode
+) -> CInt {
+#if ENABLE_MOCKING
+  if mockingEnabled { return _mock(path: path, mode) }
+#endif
+  return mkdir(path, mode)
+}
+
+internal func system_rmdir(
+    _ path: UnsafePointer<CInterop.PlatformChar>
+) -> CInt {
+#if ENABLE_MOCKING
+  if mockingEnabled { return _mock(path: path) }
+#endif
+  return rmdir(path)
```

Comment:
Similar, `RemoveDirectoryW` should be preferred.

Reply:
Somewhat asymmetrically, we're using `RemoveDirectoryW` on Windows (see `FilePathTemp.swift`), mostly because the code for `recursiveRemove` isn't shared whereas `makeDirectory` is. Though maybe `makeDirectory` should use `CreateDirectoryW` on Windows and set a security descriptor?